src/xia2/Modules/MultiCrystal/ScaleAndMerge.py

```diff
@@ -517,95 +520,56 @@ def run(self) -> None:
-
-                    _, list_of_clusters = (
-                        MultiCrystalAnalysis.interesting_cluster_identification(
-                            clusters, self._params
-                        )
-                    )
+        parallel_nproc = copy.deepcopy(self._params.nproc)
+        self._params.nproc = 1
+
+        logger.debug(
+            f"Using nproc = {parallel_nproc} for parallel scaling, PHIL nproc set to {self._params.nproc}"
+        )
 
-                    for item in list_of_clusters:
-                        cluster_no = item.split("_")[-1]
-                        cluster_dir = f"{cty}_cluster_{cluster_no}"
-
-                        for cluster in clusters:
-                            if f"cluster_{cluster.cluster_id}" == item:
-                                ids = (
-                                    self.cc_cluster_ids[cluster.cluster_id]
-                                    if k
-                                    else self.cos_cluster_ids[cluster.cluster_id]
-                                )
-                                self._scale_and_report_cluster(
-                                    self._data_manager, cluster_dir, ids, cluster
-                                )
-                                break
-        if (
-            self._params.clustering.output_clusters
-            and "coordinate" in self._params.clustering.method
+        with (
+            record_step("dials.scale(parallel)"),
+            concurrent.futures.ProcessPoolExecutor(max_workers=parallel_nproc) as pool,
         ):
-            logger.notice(banner("Scaling and merging coordinate clusters"))  # type: ignore
-            significant_clusters: list[ClusterInfo] = (
-                self._mca.significant_coordinate_clusters
-            )
-            count = 0
-            for s_c in significant_clusters:
-                if s_c.completeness < self._params.clustering.min_completeness:
-                    continue
-                if s_c.multiplicity < self._params.clustering.min_multiplicity:
-                    continue
-                if len(s_c.labels) < self._params.clustering.min_cluster_size:
-                    continue
-                if count >= self._params.clustering.max_output_clusters:
-                    continue
-                if len(s_c.labels) == len(self._data_manager.experiments):
-                    continue
-                cluster_dir = f"coordinate_cluster_{s_c.cluster_id}"
-                cluster_identifiers = [
-                    self._data_manager.ids_to_identifiers_map[l] for l in s_c.labels
-                ]
-                self._scale_and_report_cluster(
-                    self._data_manager, cluster_dir, cluster_identifiers, s_c
-                )
-                count += 1
+            cluster_futures = {
+                pool.submit(
+                    self._scale_and_report_cluster,
+                    self._data_manager,
+                    cluster[0],
```

Comment:
Indexing into a lust for the subcluster attributes looks unclear and error prone. I think we need to define a simple SubCluster dataclass with the relevant named attributes. Then `get_subclusters` will have a much simpler type signature and everything will be a bit clearer in several functions in `cluster_analysis.py`